### ISSUE TITLE
Can we use this to build whatsapp chat bot

### ISSUE THREAD

**Issue by garg-ashu** [bcalik/Whatsapp-Desktop#55]
@bcalik  I was just curious that can we build simple chat bot around this framework. Any idea...

**Comment by Enrico204** [comment]
Pre: this is not a framework, this is only a "webview" to web.whatsapp.com. Just like a browser without toolbars, etc.

It might be very difficult to build some automations. To be honest, some months ago I looked into some JS files and I made some reverse engineering in order to decode/encode packets from/to server. It was promising (some things works), but I have no time for that kind of development right now. Maybe I'll publish some source code on GitHub - I need to check if there are some personal data (eg. phone number).
Before that, my first approach was using DOM and jQuery to simulate click actions from "Developer Tool Console" (on real browser), but it didn't work (it was very unreliable).

PS: I assume that you know that build chatbots violate WhatsApp terms

**Comment by garg-ashu** [comment]
@Enrico204  Can we atleast get new incoming messages from this webview into some sort of file or string object etc.
If you can share some JS files that will be really helpful.
I know whtsapp doesn't allow you to build chatbots i am just doing it for learning purpose, no business involved.

**Comment by Enrico204** [comment]
Hi @garg-ashu , I'll try to explain some points:
- Some months ago, I tried (without success) with jQuery and DOM, injecting a JS into web.whatsapp.org, but it seems that there is no way to click to various items (such as the "item" into "chat list" for viewing messages). I was using Developer Tools of Chrome.
- After that, I tried to reverse engineer the whatsapp web protocol by reimplementing JS files from Whatsapp in Python. I have that code, I'll share it as soon as I have time :-)

My reminder about chatbots is because of banning that WhatsApp is doing on people that are using another library which is done by reverse engineering the WhatsApp android app. That library seems to work, however. Until WhatsApp bans you.
